src/librustdoc/html/render/context.rs

```diff
@@ -665,7 +665,7 @@ impl<'tcx> FormatRenderer<'tcx> for Context<'tcx> {
                 _ => unreachable!(),
             };
             let items = self.build_sidebar_items(module);
-            let js_dst = self.dst.join("sidebar-items.js");
+            let js_dst = self.dst.join(&format!("sidebar-items{}.js", self.shared.resource_suffix));
```

Comment:
@GuillaumeGomez just a thought: 

docs.rs [is using static file categorization like this](https://github.com/rust-lang/docs.rs/blob/a9d9025f84537aa47d2aedac2862b57d99d4807d/src/docbuilder/rustwide_builder.rs#L585-L588): 

```rust
let mut rustdoc_flags = vec![if create_essential_files {
    "--emit=unversioned-shared-resources,toolchain-shared-resources"
} else {
    "--emit=invocation-specific"
}
```

Was `sidebar-items.js` only missing the resource suffix? Or was/it it also in the wrong category? (correct would be `invocation-specific` as I understand it)

Reply:
sidebar-items isn't a shared resource, or even invocation-specific. It's actually *per-module*. These URLs all exist, and all have different contents:

* https://doc.rust-lang.org/stable/alloc/sync/sidebar-items.js
* https://doc.rust-lang.org/stable/alloc/sidebar-items.js
* https://doc.rust-lang.org/stable/core/sidebar-items.js

In contrast, search-index is here:

* https://doc.rust-lang.org/stable/search-index1.57.0.js

Reply:
Just like @notriddle said. ;)

Reply:
thanks for the explanation! 
So as I understand it these are invocation-specific, and when they are different they will be in different paths inside the docs.